x-pack/libbeat/reader/etw/syscall_tdh.go

```diff
@@ -0,0 +1,323 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package etw
+
+import (
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	tdh                       = windows.NewLazySystemDLL("tdh.dll")
+	tdhEnumerateProviders     = tdh.NewProc("TdhEnumerateProviders")
+	tdhGetEventInformation    = tdh.NewProc("TdhGetEventInformation")
+	tdhGetEventMapInformation = tdh.NewProc("TdhGetEventMapInformation")
+	tdhFormatProperty         = tdh.NewProc("TdhFormatProperty")
+	tdhGetProperty            = tdh.NewProc("TdhGetProperty")
+)
+
+const ANYSIZE_ARRAY = 1 << 16
```

Comment:
This is used as a const to define the maximum size for array bounds more than a variable. There are some arrays that we don't know their size beforehand. That's why I set a relatively large constant to avoid any out-of-bound errors.